android/java/org/chromium/chrome/browser/playlist/PlaylistHostActivity.java

```diff
@@ -0,0 +1,492 @@
+                                }
+                            }
+                        } else if (option == PlaylistOptions.DELETE_PLAYLIST) {
+                            if (mPlaylistService != null
+                                    && playlistOptionsModel.getPlaylistModel() != null) {
+                                mPlaylistService.removePlaylist(
+                                        playlistOptionsModel.getPlaylistModel().getId());
+                                loadAllPlaylists();
+                            }
+                        } else if (option == PlaylistOptions.MOVE_PLAYLIST_ITEMS) {
+                            showMoveOrCopyPlaylistBottomSheet();
+                        } else if (option == PlaylistOptions.COPY_PLAYLIST_ITEMS) {
+                            showMoveOrCopyPlaylistBottomSheet();
+                        }
+                    }
+                });
+
+        mPlaylistViewModel.getAllPlaylistOption().observe(
+                PlaylistHostActivity.this, playlistOptionsModel -> {
+                    if (mPlaylistService != null) {
+                        PlaylistOptions option = playlistOptionsModel.getOptionType();
+                        if (option == PlaylistOptions.REMOVE_ALL_OFFLINE_DATA) {
+                            if (mPlaylistService != null
+                                    && playlistOptionsModel.getAllPlaylistModels() != null) {
+                                for (PlaylistModel playlistModel :
+                                        playlistOptionsModel.getAllPlaylistModels()) {
+                                    mPlaylistService.removeLocalDataForItemsInPlaylist(
+                                            playlistModel.getId());
+                                }
+                            }
+                        } else if (option
+                                == PlaylistOptions.DOWNLOAD_ALL_PLAYLISTS_FOR_OFFLINE_USE) {
+                            if (mPlaylistService != null) {
+                                mPlaylistService.getAllPlaylists(playlists -> {
+                                    for (Playlist playlist : playlists) {
+                                        for (PlaylistItem playlistItem : playlist.items) {
+                                            mPlaylistService.recoverLocalDataForItem(
+                                                    playlistItem.id, true,
+                                                    tempPlaylistItem
+                                                    -> {
+
+                                                    });
+                                        }
+                                    }
+                                });
+                            }
+                        }
+                    }
+                });
+
+        mPlaylistViewModel.getPlaylistItemOption().observe(
+                PlaylistHostActivity.this, playlistItemOption -> {
+                    if (mPlaylistService != null) {
```

Comment:
maybe it just more cleaner to return if it's `null`? In that case we don't have to shift all code on one tab right.